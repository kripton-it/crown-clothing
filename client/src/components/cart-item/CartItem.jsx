import React from "react";

import {
  CartItemContainer,
  ItemsDetails,
  ItemName,
  ItemPrice
} from "./CartItemStyles";

const CartItem = ({ item }) => {
  const { imageUrl, price, name, quantity } = item;
  return (
    <CartItemContainer>
      <img src={imageUrl} alt={name} />
      <ItemsDetails>
        <ItemName>{name}</ItemName>
        <ItemPrice>{`${quantity} x $${price}`}</ItemPrice>
      </ItemsDetails>
    </CartItemContainer>
  );
};

export default CartItem;
