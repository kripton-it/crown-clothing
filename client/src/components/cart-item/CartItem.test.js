import React from "react";
import { shallow } from "enzyme";

import CartItem from "./CartItem";

describe("CartItem component", () => {
  const mockItem = {
    imageUrl: "abc.com",
    price: 100,
    name: "hello",
    quantity: 5
  };

  const wrapper = shallow(<CartItem item={mockItem} />);

  it("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("should render info:", () => {
    it("img", () => {
      const image = wrapper.find("CartItemContainer img");
      expect(image.prop("src")).toBe("abc.com");
      expect(image.prop("alt")).toBe("hello");
    });

    it("name", () => {
      const itemName = wrapper.find("ItemName");
      expect(itemName.text()).toBe("hello");
    });

    it("price", () => {
      const itemPrice = wrapper.find("ItemPrice");
      expect(itemPrice.text()).toBe("5 x $100");
    });
  });
});
