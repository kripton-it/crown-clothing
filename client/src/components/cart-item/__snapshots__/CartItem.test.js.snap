// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem component should render 1`] = `ShallowWrapper {}`;
